Sources/NIOCore/ByteBuffer-core.swift

```diff
@@ -825,6 +825,23 @@ public struct ByteBuffer {
         }
         return true
     }
+    
+    /// Shrinks the buffers capacity to a `desiredCapacity` without altering readably bytes.
```

Comment:
Let's elaborate on this by explaining the actual algorithm, just to save people from looking at the code. Something like:

```
/// The `ByteBuffer` will successfully be shrunk if the requested capacity is less than the current capacity, and the requested capacity is more than the number of readable bytes in the buffer. If either condition is not true, the buffer will not be shrunk.
```